### PR TITLE
Support for new RP `automanage`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
new resource `azurerm_automanage_configuration_profile`
new resource `azurerm_automanage_configuration_profile_assignment`
new resource `azurerm_automanage_configuration_profile_hci_assignment`

### CODE REVIEW

**Comment by liuwuliuyun** [comment]
Testing evidence on Teamcity will be provided shortly.

**Comment by liuwuliuyun** [comment]
image

File: website/docs/r/automanage_configuration_profiles.html.markdown [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+resource "azurerm_resource_group" "example" {
+  name     = "example-automanage"
+  location = "West Europe"
+}
+
+resource "azurerm_automanage_configuration_profile" "example" {
+  name                = "example-configurationprofile"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  configuration       = "{\"Antimalware/Enable\":false,\"AzureSecurityCenter/Enable\":true,\"Backup/Enable\":false,\"BootDiagnostics/Enable\":true,\"ChangeTrackingAndInventory/Enable\":true,\"GuestConfiguration/Enable\":true,\"LogAnalytics/Enable\":true,\"UpdateManagement/Enable\":true,\"VMInsights/Enable\":true}"
```
Comment (katbyte) [inline website/docs/r/automanage_configuration_profiles.html.markdown:25]: this is json so would it be better as
```suggestion
  configuration_json       = "{\"Antimalware/Enable\":false,\"AzureSecurityCenter/Enable\":true,\"Backup/Enable\":false,\"BootDiagnostics/Enable\":true,\"ChangeTrackingAndInventory/Enable\":true,\"GuestConfiguration/Enable\":true,\"LogAnalytics/Enable\":true,\"UpdateManagement/Enable\":true,\"VMInsights/Enable\":true}"
```

File: website/docs/r/automanage_configuration_profile_hci_assignments.html.markdown [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  location            = azurerm_resource_group.test.location
+  client_id           = data.azurerm_client_config.current.client_id
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+}
+
+resource "azurerm_automanage_configuration_profile_hci_assignment" "test" {
+  name                  = "default"
+  resource_group_name   = azurerm_resource_group.test.name
+  cluster_name          = azurerm_stack_hci_cluster.test.name
+  configuration_profile = "/providers/Microsoft.Automanage/bestPractices/AzureBestPracticesProduction"
```
Comment (katbyte) [inline website/docs/r/automanage_configuration_profile_hci_assignments.html.markdown:39]: this is an id?
```suggestion
  configuration_profile_id = "/providers/Microsoft.Automanage/bestPractices/AzureBestPracticesProduction"
```

File: website/docs/r/automanage_configuration_profile_assignments.html.markdown [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this automanage ConfigurationProfileAssignment. Changing this forces a new automanage ConfigurationProfileAssignment to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the automanage ConfigurationProfileAssignment should exist. Changing this forces a new automanage ConfigurationProfileAssignment to be created.
+
+* `vm_name` - (Required) The name of the virtual machine. Changing this forces a new automanage ConfigurationProfileAssignment to be created.
+
+* `configuration_profile` - (Required) The Automanage configurationProfile ARM Resource URI.
```
Comment (katbyte) [inline website/docs/r/automanage_configuration_profile_assignments.html.markdown:123]: this is an id?
```suggestion
* `configuration_profile_id` - (Required) The Automanage configurationProfile ARM Resource URI.
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/automanage_configuration_profiles.html.markdown (hunk 1 above)
Comment (liuwuliuyun) [reply website/docs/r/automanage_configuration_profiles.html.markdown:25]: Done

File: website/docs/r/automanage_configuration_profile_hci_assignments.html.markdown (hunk 2 above)
Comment (liuwuliuyun) [reply website/docs/r/automanage_configuration_profile_hci_assignments.html.markdown:39]: Done

File: website/docs/r/automanage_configuration_profile_assignments.html.markdown (hunk 3 above)
Comment (liuwuliuyun) [reply website/docs/r/automanage_configuration_profile_assignments.html.markdown]: Done

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @liuwuliuyun. There are a number of issues that need to be fixed up in addition to the ones I've left in line:

1. These need to be typed resources, because these will have to be typed resources in the future
2. The create and update methods should be split and take advantage of `d.HasChanges()`, so that ignore changes works
3. We need to remove the `managed_by` and `type` properties since they're not something users should need to set or care about
4. The `configuration_json` property in `azurerm_automanage_configuration_profile` needs to be exposed as individual properties or a block of properties if it makes sense to group them, could you provide some examples of how this is supposed to be used?

Once the points above have been addressed we can take another look through this.

Thanks!

**Comment by liuwuliuyun** [comment]
Thanks for the comment, I will work on the improvment and let you know the progress.

**Comment by liuwuliuyun** [comment]
Hi @katbyte and @stephybun, I agree that swiching to Typed SDK is good for future update. But switching to typed sdk will result in a lot code change, and make it hard to review. So I will close this PR for now and open a new one after switching.

**Pull request closed without merging**